packages/core/src/asset/AssetPromise.ts

```diff
@@ -173,3 +182,30 @@ enum PromiseState {
   Rejected = "rejected",
   Canceled = "canceled"
 }
+
+/**
+ * Progress interface.
+ */
+export interface IProgress {
+  /**
+   * Progress detail.
+   * @remarks
+   * The key is the asset url, the value is the progress of the asset url.
+   * Some assets may have multiple urls, such as glTF, which has a glTF file and multiple texture files.
+   */
+  detail?: {
```

Comment:
How do I direct which asset the current progress trigger is? Three and babylon have